src/test/java/uk/gov/pay/directdebit/pact/PublicApiContractTest.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.pay.directdebit.pact;
+
+import au.com.dius.pact.provider.junit.PactRunner;
+import au.com.dius.pact.provider.junit.Provider;
+import au.com.dius.pact.provider.junit.State;
+import au.com.dius.pact.provider.junit.loader.PactBroker;
+import au.com.dius.pact.provider.junit.loader.PactBrokerAuth;
+import au.com.dius.pact.provider.junit.target.HttpTarget;
+import au.com.dius.pact.provider.junit.target.Target;
+import au.com.dius.pact.provider.junit.target.TestTarget;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.runner.RunWith;
+import uk.gov.pay.directdebit.junit.DropwizardAppWithPostgresRule;
+import uk.gov.pay.directdebit.payments.fixtures.GatewayAccountFixture;
+
+import java.util.Map;
+
+@RunWith(PactRunner.class)
+@Provider("direct-debit-connector")
+@PactBroker(protocol = "https", host = "pact-broker-test.cloudapps.digital", port = "443", tags = {"${PACT_CONSUMER_TAG}"},
+        authentication = @PactBrokerAuth(username = "${PACT_BROKER_USERNAME}", password = "${PACT_BROKER_PASSWORD}"))
+//@PactFolder("pacts") <-- this is useful for testing pacts locally
+public class PublicApiContractTest {
+
+    @ClassRule
+    public static DropwizardAppWithPostgresRule app = new DropwizardAppWithPostgresRule();
+    
+    @TestTarget
+    public static Target target;
+
+    private GatewayAccountFixture testGatewayAccount = GatewayAccountFixture.aGatewayAccountFixture();
+
+    @BeforeClass
+    public static void setUpService() {
+        target = new HttpTarget(app.getLocalPort());
+    }
+    
+    @State("a gateway account with external id exists")
+    public void aGatewayAccountWithExternalIdExists(Map<String, String> params) {
+        testGatewayAccount.withExternalId(params.get("gateway_account_id")).insert(app.getTestContext().getJdbi());
+    }
+}
```

Comment:
It would be good to find some way of being able to have separate pact test files for different bits of functionality, otherwise this file will become huge and unmanageable.

Reply:
this was one of the problems at my last job! The provider test class was getting huge. I don't know if there's a way to separate it out just now. I'll ask in the pact slack but I don't think it's an issue that should prevent this PR from progressing at the moment as the provider test class still looks quite clean.

Reply:
ok cool. Would be good to figure this out soonish